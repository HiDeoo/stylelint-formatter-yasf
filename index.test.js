const formatter = require('.')

const defaultResult = {
  source: 'path/to/file.css',
  warnings: [],
  deprecations: [],
  invalidOptionWarnings: [],
}

const defaultError = {
  line: 3,
  column: 12,
  rule: 'block-no-empty',
  severity: 'error',
  text: 'You should not have an empty block (block-no-empty)',
}

let actualTTY
let actualColumns

beforeAll(() => {
  actualTTY = process.stdout.isTTY
  actualColumns = process.stdout.columns
})

beforeEach(() => {
  process.stdout.columns = 80
})

afterEach(() => {
  process.stdout.isTTY = actualTTY
  process.stdout.columns = actualColumns
})

test('should output nothing when not necessary', () => {
  const results = [defaultResult]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output invalid options', () => {
  const results = [
    {
      ...defaultResult,
      invalidOptionWarnings: [
        {
          text: 'Invalid option X for rule Y',
        },
        {
          text: 'Invalid option A for rule B',
        },
        {
          text: 'Invalid option X for rule Y',
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output deprecations', () => {
  const results = [
    {
      ...defaultResult,
      deprecations: [
        {
          text: 'Feature X has been deprecated and will be removed in the next major version.',
        },
        {
          text: 'Feature Y has been deprecated and will be removed in the next major version.',
        },
        {
          text: 'Feature X has been deprecated and will be removed in the next major version.',
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output 1 warning', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [defaultError],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output multiple warnings', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        defaultError,
        {
          ...defaultError,
          line: 44,
          column: 22,
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output warning without TTY', () => {
  process.stdout.isTTY = false

  const results = [
    {
      ...defaultResult,
      warnings: [defaultError],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should not output symbols for unknown sevirity', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        {
          ...defaultError,
          severity: 'test',
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output warnings and errors', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        defaultError,
        {
          ...defaultError,
          severity: 'warning',
          line: 44,
          column: 22,
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output long warning when less than 80 columns are available', () => {
  process.stdout.columns = 60

  const results = [
    {
      ...defaultResult,
      warnings: [
        {
          ...defaultError,
          text:
            'A very very very very very very very very very very very very very very very very very very long error',
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output long warning when more than 80 columns are available', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        {
          ...defaultError,
          text:
            'A very very very very very very very very very very very very very very very very very very long error',
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should not output file names if not provided', () => {
  const results = [
    {
      ...defaultResult,
      source: null,
      warnings: [defaultError],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should not output lines & columns if not provided', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        {
          ...defaultError,
          line: null,
          column: 22,
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should not output columns if not provided', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        {
          ...defaultError,
          column: null,
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should output parseErrors as warnings', () => {
  const results = [
    {
      ...defaultResult,
      parseErrors: [defaultError],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should not try to get the relative path if starting with "<"', () => {
  const results = [
    {
      ...defaultResult,
      source: '<path/to/file.css>',
      warnings: [defaultError],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()
})

test('should properly order errors on the same line', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        defaultError,
        {
          ...defaultError,
          column: 22,
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()

  results[0].warnings[1].column = 3

  expect(formatter(results)).toMatchSnapshot()
})

test('should properly order errors on different lines', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        defaultError,
        {
          ...defaultError,
          line: 4,
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()

  results[0].warnings[1].line = 1

  expect(formatter(results)).toMatchSnapshot()
})

test('should properly order errors and warnings', () => {
  const results = [
    {
      ...defaultResult,
      warnings: [
        defaultError,
        {
          ...defaultError,
          line: 1,
        },
      ],
    },
  ]

  expect(formatter(results)).toMatchSnapshot()

  results[0].warnings[1].severity = 'warning'

  expect(formatter(results)).toMatchSnapshot()

  results[0].warnings[0].severity = 'warning'
  results[0].warnings[1].severity = 'error'

  expect(formatter(results)).toMatchSnapshot()
})
