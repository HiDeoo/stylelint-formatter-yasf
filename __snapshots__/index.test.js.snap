// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not output columns if not provided 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should not output file names if not provided 1`] = `
"
[2m3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should not output lines & columns if not provided 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should not output symbols for unknown sevirity 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m     You should not have an empty block   [2mblock-no-empty[22m

"
`;

exports[`should not try to get the relative path if starting with "<" 1`] = `
"
[4m<path/to/file.css>[24m
 [2m<path/to/file.css>:3:12[22m  [31mâœ–[39m  You should not have an empty block  [2mblock-no-empty[22m 
                                                                                
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should output 1 warning 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should output deprecations 1`] = `
"
[33mDeprecated rule: [39mFeature X has been deprecated and will be removed in the next major version.
[33mDeprecated rule: [39mFeature Y has been deprecated and will be removed in the next major version.
[4m[24m
[4mSummary:[24m[33m[39m
[33m [34mâ„¹[33m 2 deprecations[39m

"
`;

exports[`should output invalid options 1`] = `
"
[31mInvalid option: [39mInvalid option X for rule Y.
[31mInvalid option: [39mInvalid option A for rule B.
[4m[24m
[4mSummary:[24m[31m[39m
[31m [34mâ„¹[31m 2 invalid options[39m

"
`;

exports[`should output long warning when less than 80 columns are available 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  A very very very very very very very  [2mblock-no-empty[22m 
                           very very very very very very very v                 
                           ery very very very long error                        
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should output long warning when more than 80 columns are available 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  A very very very very very very very  [2mblock-no-empty[22m 
                           very very very very very very very v                 
                           ery very very very long error                        
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should output multiple warnings 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m   [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:44:22[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 2 errors[39m

"
`;

exports[`should output nothing when not necessary 1`] = `""`;

exports[`should output parseErrors as warnings 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block (block-no-empt    
                           y)                                                   
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should output warning without TTY 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m

"
`;

exports[`should output warnings and errors 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m   [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:44:22[22m  [33mâš [39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m[33m[39m
[33m [33mâš [33m 1 warning[39m

"
`;

exports[`should properly order errors and warnings 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:1:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 2 errors[39m

"
`;

exports[`should properly order errors and warnings 2`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:1:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:3:12[22m  [33mâš [39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m[33m[39m
[33m [33mâš [33m 1 warning[39m

"
`;

exports[`should properly order errors and warnings 3`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:1:12[22m  [33mâš [39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 1 error[39m[33m[39m
[33m [33mâš [33m 1 warning[39m

"
`;

exports[`should properly order errors on different lines 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:4:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 2 errors[39m

"
`;

exports[`should properly order errors on different lines 2`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:1:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 2 errors[39m

"
`;

exports[`should properly order errors on the same line 1`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:3:22[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 2 errors[39m

"
`;

exports[`should properly order errors on the same line 2`] = `
"
[4mpath/to/file.css[24m
 [2mpath/to/file.css:3:3[22m   [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
 [2mpath/to/file.css:3:12[22m  [31mâœ–[39m  You should not have an empty block   [2mblock-no-empty[22m 
[4m[24m
[4mSummary:[24m[31m[39m
[31m [31mâœ–[31m 2 errors[39m

"
`;
